packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -0,0 +1,262 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show Offset;
+
+import 'package:flutter/foundation.dart';
+
+import 'arena.dart';
+import 'constants.dart';
+import 'events.dart';
+import 'recognizer.dart';
+
+enum _ForceState {
+  // No pointer has touched down and the detector is ready for a pointer down to occur.
+  ready,
+
+  // A pointer has touched down, but a force press gesture has not yet been detected.
+  possible,
+
+  // A pointer is down and a force press gesture has been detected. However, if
+  // ForcePressGestureRecognizer the only recognizer in the arena, the gesture
+  // will not yet have started.
+  accepted,
+
+  // A pointer is down and the gesture has started, ie. the pressure of the pointer
+  // has just become greater than the ForcePressGestureRecognizer.startPressure.
+  started,
+
+  // A pointer is down and the pressure of the pointer has just become greater
+  // than the ForcePressGestureRecognizer.peakPressure. Even after a pointer
+  // crosses this threshold, onUpdate callbacks will still be sent.
+  peaked,
+}
+
+/// Details object for callbacks that use [GestureForcePressStartCallback],
+/// [GestureForcePressPeakCallback], [GestureForcePressEndCallback] or
+/// [GestureForcePressUpdateCallback].
+///
+/// See also:
+///
+///  * [ForcePressGestureRecognizer.onStart], [ForcePressGestureRecognizer.onPeak],
+///    [ForcePressGestureRecognizer.onEnd], and [ForcePressGestureRecognizer.onUpdate]
+///    which use [ForcePressDetails].
+///  * [ForcePressUpdateDetails], the details for [ForcePressUpdateCallback].
+class ForcePressDetails {
+  /// Creates details for a [GestureForcePressStartCallback],
+  /// [GestureForcePressPeakCallback] or [GestureForcePressEndCallback].
+  ///
+  /// The [globalPosition] argument must not be null.
+  ForcePressDetails({
+    @required this.globalPosition,
+    @required this.pressure,
+  }) : assert(globalPosition != null),
+       assert(pressure != null);
+
+  /// The global position at which the function was called.
+  final Offset globalPosition;
+
+  /// The pressure of the pointer on the screen.
+  final double pressure;
+}
+
```

Comment:
It's fine to do it in a different PR but I was saying above, there will be cases where users will use the pressure double value before start is triggered. See on the home screen, when you barely force press an icon, it'll visually respond before you trigger the start.